packages/rum-core/src/domain/resource/resourceUtils.ts

```diff
@@ -151,39 +150,15 @@ export function toValidEntry(entry: RumPerformanceResourceTiming) {
       entry.requestStart,
       entry.responseStart,
       entry.responseEnd
-    )
+    ) &&
+    (!hasRedirection(entry) || areInOrder(entry.startTime, entry.redirectStart, entry.redirectEnd, entry.fetchStart))
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏The second part of the condition is a bit hard to get at the first reading. Maybe doing it like so would ease the understanding 🤷‍♂️

```js
if (
    !areInOrder(
      entry.startTime,
      entry.fetchStart,
      entry.domainLookupStart,
      entry.domainLookupEnd,
      entry.connectStart,
      entry.connectEnd,
      entry.requestStart,
      entry.responseStart,
      entry.responseEnd
    ) ||
    (hasRedirection(entry) && !areInOrder(entry.startTime, entry.redirectStart, entry.redirectEnd, entry.fetchStart)) {
    return 
}
```